[Hester James Douglas] iP

e0543403 opened the pull request:
# **Duke**

> Life is too complicated not to be orderly. [(source)](https://dansilvestre.com/productivity-quotes/#10_Life_is_too_complicated_not_to_be_orderly_-_Martha_Stewart)

Duke helps to remember things you need to do to keep your life orderly. It is,

- text-based
- easy to learn
- ~fast~ Super Fast to use

All you need to do is,

1. download it from [here](https://github.com/e0543403/ip/releases/tag/v0.1).
2. put it in your preferred folder
3. launch it via command line 😅  typing  java -jar duke.java
4. let it manage your tasks for you

Did I mention its **FREE**?!

Features:

- [x] Manages tasks
- [ ]  Organizes tasks (coming eventually)
- [ ]  Reminders (coming eventually)


If you are a Java programmer you can use it to practice Java too. Here is the `main` method:
```java
public static void main(String[] args) {
    String PATH = System.getProperty("user.dir");

    new Duke(PATH).run();
}
```

Commit:
Add Gradle support

phantl committed:
Add greeting

phantl committed:
Add echo and exit

phantl committed:
Add DukeList.java

phantl committed:
Add add to DukeList.java

phantl committed:
Add list to DukeList.java

phantl committed:
Replace echo with add in Duke.java

phantl committed:
Add list to Duke.java

phantl committed:
Add Task.java

phantl committed:
Add toString functionality to Task.java

phantl committed:
Add done to Task.java

phantl committed:
Replace String list with Task list in DukeList.java

phantl committed:
Add done to DukeList.java

phantl committed:
Remove output from done in Task.java

phantl committed:
Add done to Duke.java

phantl committed:
Add Events.java

phantl committed:
Add Deadlines.java

phantl committed:
Add ToDos.java

phantl committed:
Add addTodo in DukeList.java

phantl committed:
Add addDeadlines to DukeList.java

phantl committed:
Add addEvents to DukeList.java

phantl committed:
Add spacing to toString in Events.java

phantl committed:
Add spacing to toString in Deadlines.java

phantl committed:
Add spacing to displayTask in DukeList.java

phantl committed:
Add todo to Duke.java

phantl committed:
Add deadline to Duke.java

phantl committed:
Add events to Duke.java

phantl committed:
Add test case to input.txt and EXPECTED.TXT

phantl committed:
Add DukeException.java

phantl committed:
Change input of parameters for Duke.java

phantl committed:
Add trimming to addTodo addDeadlines addEvents in DukeList.java

phantl committed:
Add throw DukeException to addTodo in DukeList.java

phantl committed:
Add try catch to Duke.java for addTodo

phantl committed:
Add additional tests for Duke.java in input.txt and EXPECTED.TXT

phantl committed:
Change try catch to encompass all if else statements

phantl committed:
Change add in Duke.java to throw a DukeException always

phantl committed:
Add more tests for Duke.java to input.txt and EXPECTED.TXT

phantl committed:
Replace task array with task arraylist

phantl committed:
Add delete() to DukeList.java

phantl committed:
Add delete functionality to Duke.java

phantl committed:
Add fullstops to taskCount strings

phantl committed:
Add more tests for Duke.java in input.txt and EXPECTED.TXT

phantl committed:
Change name for addTodo method to addToDo

phantl committed:
Add javadocs to Task.java

e0543403 committed:
Add javadocs to ToDos.java

e0543403 committed:
Add javadocs to Deadlines.java

e0543403 committed:
Add javadocs to Events.java

e0543403 committed:
Add javadocs to DukeException.java

e0543403 committed:
Add javadocs to DukeList.java

e0543403 committed:
Add skeleton class Storage.java to java directory

e0543403 committed:
Add missing line in Storage.java constructor

e0543403 committed:
Import java FileReader, FileWriter, BufferedReader, IOException

e0543403 committed:
Implement save method in Storage.java

e0543403 committed:
Add saveData method to Task.java and its child classes

e0543403 committed:
Add toString method to DukeList.java

e0543403 committed:
Add loadData method to DukeList.java

e0543403 committed:
Add load method functionality and closed fileWriter in save method

e0543403 committed:
Shift Strings in main into Duke.java

e0543403 committed:
Add save and load functionality to Duke.java

e0543403 committed:
Add data file and data.txt file for saving and loading data

e0543403 committed:
Shift location of data.txt

e0543403 committed:
Add ability to create file and directory for saved data if none exist

e0543403 committed:
Merge branch 'branch-Level-7'

e0543403 committed:
Add data.txt and data directory to gitignore

e0543403 committed:
Replace String limit with LocalDate object

e0543403 committed:
Add throw DukeException for addDeadlines method in DukeList.java

e0543403 committed:
Remove extra data.txt file

e0543403 committed:
Add more tests to input.txt and EXPECTED.TXT for Duke.java

e0543403 committed:
Add javadocs to Storage.java

e0543403 committed:
Merge branch 'branch-Level-8'

e0543403 committed:
Add skeleton classes Parser.java and Ui.java

e0543403 committed:
Add parse method to Parser.java

e0543403 committed:
Add run method to Ui.java

e0543403 committed:
Remove statement of throwing IOException

e0543403 committed:
Replace Duke main method functionality with MoreOOP classes

e0543403 committed:
Shift all Duke classes into packages

e0543403 committed:
Add ToDosTest.java test class in test folder

e0543403 committed:
Repair javadocs of classes in task package

e0543403 committed:
Add EventsTest.java test class in test folder

e0543403 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

e0543403 committed:
Set correct mainClassName in build.gradle

e0543403 committed:
Change saving to be done after changes in list

e0543403 committed:
Add JavaDocs for Parser.java

e0543403 committed:
Add JavaDocs for Ui.java

e0543403 committed:
Add JavaDocs for Deadlines.java

e0543403 committed:
Add JavaDocs for DukeList.java

e0543403 committed:
Add JavaDocs for Events.java

e0543403 committed:
Add JavaDocs for Task.java

e0543403 committed:
Add JavaDocs for ToDos.java

e0543403 committed:
Add JavaDocs for Duke.java

e0543403 committed:
Reorder import statements

e0543403 committed:
Change name of arraylist collection to plural noun

e0543403 committed:
Capitalize all constant names with underscore to seperate words

e0543403 committed:
Add find method to DukeList.java

e0543403 committed:
Add find function to Parser.java parse method

e0543403 committed:
Merge branch 'branch-Level-9'

e0543403 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/task/DukeList.java

Resolution: keep both

e0543403 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Ui.java
#	src/main/java/duke/task/DukeList.java

Resolution: keep both

e0543403 committed:
Add JavaDoc for find method in DukeList.java

e0543403 committed:
Fix runtest.bat and test file expected outputs

e0543403 committed:
Add checkstyle files to iP

e0543403 committed:
Change style of programs to follow checkstyle

e0543403 committed:
Change test files to follow checkstyle

e0543403 committed:
Add JavaFX dependencies to build.gradle

e0543403 committed:
Add Launcher class for GUI functionality

e0543403 committed:
Add JavaFX GUI to Duke

e0543403 committed:
Change return type of methods involving the list

The commands previously directly printed onto the terminal.

Therefore to work with the current GUI they had to be changed to be
able to pass them into the GUI response system.

e0543403 committed:
Update Javadocs of methods

e0543403 committed:
Update format to adhere to checkstyle

e0543403 committed:
Change main class name to Launcher

e0543403 committed:
Fix functionality lost in Ui

e0543403 committed:
Remove additional function call

e0543403 committed:
Merge branch 'branch-Level-10'

e0543403 committed:
Add tweaks to GUI

dannytayjy commented on src/main/java/duke/task/DukeList.java:
```diff
@@ -0,0 +1,234 @@
+     * @param text Description of the task to be added.
+     */
+    public void add(String text) {
+
```
Extra empty line?

dannytayjy replied:
Extra blank line?

dannytayjy commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,57 @@
+            + "|  _ \\ _   _| | _____ \n"
+            + "| | | | | | | |/ / _ \\\n"
+            + "| |_| | |_| |   <  __/\n"
+            + "|____/ \\__,_|_|\\_\\___|\n";
```
Missing blank line separation?

dannytayjy commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,59 @@
+    /**
+     * Marks the task as done.
+     */
+    public void done() {
```
Setter methods for boolean variables must be of the form of `setDone()`

dannytayjy commented on src/main/java/duke/task/DukeList.java:
```diff
@@ -0,0 +1,234 @@
+     * @param text Body of the task to be added.
+     * @return The message associated to the task being displayed.
+     */
+    public String addDeadlines(String text) throws DukeException {
```
Probably `addDeadline` is better?

_Plural form should be used on names representing a collection of objects._

dannytayjy commented on src/main/java/duke/task/DukeList.java:
```diff
@@ -0,0 +1,234 @@
+     * @param text Body of the task to be added.
+     * @return The message associated to the task being displayed.
+     */
+    public String addEvents(String text) {
```
Probably `addEvent` is better?

_Plural form should be used on names representing a collection of objects._

dannytayjy commented on src/main/java/duke/task/Events.java:
```diff
@@ -0,0 +1,40 @@
+/**
+ * Encapsulates a task to be done at a specified time.
+ */
+public class Events extends Task {
```
You might want to rename the class from `Events` to `Event` since the class represents an `Event` task.

_Plural form should be used on names representing a collection of objects._

dannytayjy commented on src/main/java/duke/task/ToDos.java:
```diff
@@ -0,0 +1,37 @@
+/**
+ * Encapsulates a task to be done.
+ */
+public class ToDos extends Task {
```
You might want to rename the class from `ToDos` to `ToDo` since the class represents a `ToDo` task.

_Plural form should be used on names representing a collection of objects._

e0543403 replied:
Thanks for the work mate! I've actually been wondering whether they should be plural due to  week 2 level 4's way of introducing the task.

dannytayjy reviewed (commented):
Overall well done! I did not really notice any coding violations except for one setter method and the use of plural forms.

e0543403 committed:
Fix coding standard violations

e0543403 committed:
Change name of test classes to match main classes

ruthpohrp commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,97 @@
+            fileWriter.write(list.toString());
+            fileWriter.close();
+        } catch (IOException e) {
+            System.out.println(e.toString());
```
May want to make a DukeException for this and a custom message?

ruthpohrp commented on src/main/java/duke/task/DukeList.java:
```diff
@@ -0,0 +1,233 @@
+    public String list() {
+        String str = "Here are the tasks in your list:\n";
+        for (int i = 0; i < tasks.size(); i++) {
+            str += (i + 1) + "." + tasks.get(i).toString() + "\n";
```
Try using StringBuilder for concatenating long strings like this? If list of tasks is 100 tasks long, iirc using + requires a lot of computing power or smth

ruthpohrp commented on src/main/java/duke/task/DukeList.java:
```diff
@@ -0,0 +1,233 @@
+        return str;
+    }
+
+
```
May want to check line spacing between methods? Here there's two line spaces but for some methods above there's only one in between methods

ruthpohrp commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,97 @@
+            System.out.println(e.getMessage());
+        }
+    }
+}
```
I like the spacing, very easy to read

e0543403 replied:
Thanks mate for the work you put in! I'll fix these issues as soon as possible.

ruthpohrp reviewed (commented):
For PR Review. Thanks

e0543403 committed:
Improve GUI aesthetically

e0543403 committed:
Fix checkstyle violations

e0543403 committed:
Add Image to GUI

e0543403 committed:
Fix exit function for GUI

e0543403 committed:
Fix code based on code quality guidelines

e0543403 committed:
Improve readability of find method's output

e0543403 committed:
Parser parse method: extract if else booleans

The if else conditions have similarities in checks.

This results in code duplication.

Extracting the conditions into methods allows for better readability.
Overall improving code quality.

Let's refactor the if else conditions into methods correlating to each
variant.

e0543403 committed:
Improve code quality

e0543403 committed:
DukeList methods: handle fringe cases

Delete command does not handle the user inputing an out of list number.
Deadline command does not handle the user not inputing parameters.

Duke should be able to continue running even if the user keys in
incorrect inputs.

Let's,
* update delete to handle out of list inputs
* update deadline to handle lack of parameters

e0543403 committed:
Enable assertions

e0543403 committed:
Add assertion to loadData method

e0543403 committed:
Merge pull request #1 from e0543403/branch-A-CodeQuality

Branch A Code Quality

e0543403 committed:
Merge branch 'master' of https://github.com/e0543403/ip

e0543403 committed:
Merge branch 'master' into branch-A-Assertions

e0543403 committed:
Merge pull request #2 from e0543403/branch-A-Assertions

Branch A Assertions

e0543403 committed:
Add undo functionality to Duke

It is difficult or tedious for users to undo mistakes or commands.

Adding a command to streamline the process improves user-friendliness.

Let's add an undo command to undo the last command entered.

e0543403 committed:
Change EpicFace.png

e0543403 committed:
Add DeadlineTest test class

e0543403 committed:
Change Duke's image

e0543403 committed:
Add Ui.png

e0543403 committed:
Change file paths to ensure they function cross platform

e0543403 committed:
Add user guide for Duke

e0543403 committed:
Add Duke's name to the user guide

e0543403 committed:
Correct explanation for undo command

e0543403 committed:
Add credit for reuse from stackOverflow answer

e0543403 committed:
Add additional newline for formatting
